docs/decisions/006-open-api-dynamic-payload-and-namespaces.md

```diff
@@ -0,0 +1,79 @@
+
+<!-- This is an optional element. Feel free to remove. -->
+## Decision Drivers
+* Create a mechanism that enables the dynamic construction of the payload/body for PUT and POST RestAPI operations.
+* Develop a mechanism(namespacing) that allows differentiation of payload properties with identical names at various levels for PUT and POST RestAPI operations.
+* Aim to minimize breaking changes and maintain backward compatibility of the code as much as possible.
+
+## Considered Options
+* Enable the dynamic creation of payload and/or namespacing by default.
+* Enable the dynamic creation of payload and/or namespacing based on configuration.
+
+## Decision Outcome
+Chosen option: "Enable the dynamic creation of payload and/or namespacing based on configuration". This option keeps things compatible, so the change won't affect any SK consumer code. Additionally, it lets SK consumer code easily control both mechanisms, turning them on or off based on the scenario.
+
+## Additional details
+
+### Enabling dynamic creation of payload
+In order to enable the dynamic creation of payloads/bodies for PUT and POST RestAPI operations, please set the `BuildOperationPayloadDynamically` property of the `OpenApiSkillExecutionParameters` execution parameters to `true` when importing the AI plugin:
+
+```csharp
+var plugin = await kernel.ImportAIPluginAsync("<skill name>", new Uri("<chatGPT-plugin>"), new OpenApiSkillExecutionParameters(httpClient) { BuildOperationPayloadDynamically = true });
+```
+
+To dynamically construct a payload for a RestAPI operation that requires payload like this:
+```json
+{
+	"value": "secret-value",
+	"attributes": {
+		"enabled": true
+	}
+}
+```
+
+Please register the following arguments in context variables collection:
+
+```csharp
+var contextVariables = new ContextVariables();
+contextVariables.Set("value", "secret-value");
+contextVariables.Set("enabled", true);
+```
+
+### Enabling namespacing
+To enable namespacing, set the `NamespacePayloadParameters` property of the `OpenApiSkillExecutionParameters` execution parameters to `true` when importing the AI plugin:
+
+```csharp
+var plugin = await kernel.ImportAIPluginAsync("<skill name>", new Uri("<chatGPT-plugin>"), new OpenApiSkillExecutionParameters(httpClient) { NamespacePayloadParameters = true });
```

Comment:
Enabling namespacing by default will impact existing SK consumer code that uses OpenAPI functionality directly(non- planner scenario) - importing the skill/plugin and running it through the `kernel.RunAsync(...)` method. 

Currently, to execute an OpenAPI operation, the consumer code adds operation arguments to the context variable collection using parameter names as they are declared in the OpenAPI document - without namespaces. If we enable namespacing by default, that code will fail because it expects arguments with namespaced names, while the arguments in the context variable collection have regular names. 

So, the property is false by default, and in case the SK consumer code faces an issue with property name collisions, they will activate namespacing and refactor their code to supply namespaced arguments.